### PR TITLE
Lookmanoapipath

### PR DESCRIPTION
**Pull request by sneridagh**
PoC, dev stays more or less than before (seamless by default). Production can run RAZZLE_API_PATH-less :) using Pyramid-like approach using `Host` header to "know" the server name. We could even use it for non-root deployments in the near future.

Nginx config used:

```
# This defines in which IP and port Plone is running.
# The default is 127.0.0.1:8080
upstream backend {
    server host.docker.internal:8080;
}
upstream frontend {
    server host.docker.internal:3000;
}

# server {
#   listen 80;
#   server_name localhost;
#   return 301 https://$server_name$request_uri;
# }

server {
  listen 80;
  server_name local.kitconcept.io;

  client_max_body_size 1G;

  access_log /var/log/nginx/localhost.access.log;
  error_log /var/log/nginx/localhost.error.log;

  location ~ /(.*)$ {
    location ~* \.(js|jsx|css|less|swf|eot|ttf|otf|woff|woff2)$ {
        add_header Cache-Control "public";
        expires +1y;
        proxy_pass http://frontend;
    }
    location ~* static.*\.(ico|jpg|jpeg|png|gif|svg)$ {
        add_header Cache-Control "public";
        expires +1y;
        proxy_pass http://frontend;
    }

    if ($http_accept = 'application/json') {
        rewrite ^(.*) /VirtualHostBase/http/local.kitconcept.io/Plone/VirtualHostRoot$1 break;
        proxy_pass http://backend;
        break;
    }

    proxy_set_header        Host $host;
    proxy_set_header        X-Real-IP $remote_addr;
    proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header        X-Forwarded-Proto $scheme;
    proxy_pass              http://frontend;
  }
}

```

Docker
`docker run --rm --name nginxserver -p 80:80 -v $(shell pwd)/nginx.conf:/etc/nginx/conf.d/plone.conf:ro nginx`

### CODE REVIEW

File: src/server.jsx [hunk 1]
```diff
@@ -164,6 +167,13 @@ server.get('/*', (req, res) => {
   const url = req.originalUrl || req.url;
   const location = parseUrl(url);
 
+  const hostname = `${req.protocol}://${req.headers.host}`;
+
+  if (hostname) {
```
Comment (tiberiuichim) [inline src/server.jsx:172]: Is the condition really needed?

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,8 @@
 
 ### Bugfix
 
+- Fix storybook initial config registry setup @sneridagh
```
Comment (tiberiuichim) [inline CHANGELOG.md:11]: The changelog entry is weird

File: src/helpers/Api/Api.js [hunk 3]
```diff
@@ -23,8 +23,8 @@ function formatUrl(path) {
   let apiPath = '';
   if (settings.internalApiPath && __SERVER__) {
     apiPath = settings.internalApiPath;
-  } else {
-    apiPath = settings.apiPath;
+  } else if (config.settings.apiPath) {
+    apiPath = config.settings.apiPath;
```
Comment (tiberiuichim) [inline src/helpers/Api/Api.js:27]: there's no catch-all fallback here.

File: src/helpers/Api/APIResourceWithAuth.js [hunk 4]
```diff
@@ -16,14 +16,14 @@ import config from '@plone/volto/registry';
...
   new Promise((resolve, reject) => {
     const { settings } = config;
-    let apiPath = '';
+    let backendAPIPath = '';
     if (settings.internalApiPath && __SERVER__) {
-      apiPath = settings.internalApiPath;
+      backendAPIPath = settings.internalApiPath;
     } else {
-      apiPath = settings.apiPath;
+      backendAPIPath = settings.backendAPIPath;
```
Comment (tiberiuichim) [inline src/helpers/Api/APIResourceWithAuth.js:23]: where is settings.backendAPIPath declared?

**Comment by sneridagh** [comment]
Yeah the commits are messed up don't know why, a bit in a hurry all together 😜 I have to cleanup a bit!

**Comment by sneridagh** [comment]
closed in favor of: https://github.com/plone/volto/pull/2320

**Pull request closed without merging**